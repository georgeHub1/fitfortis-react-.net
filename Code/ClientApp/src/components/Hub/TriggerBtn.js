import React from 'react';
import { Icon } from 'antd';

import styles from './styles.module.less';

const TriggerBtn = ({ onClick, collapsed }) => {
  return (
    <div className={ `${collapsed ? styles.rotateContainer : '' } ${styles.triggerContainer}` }>
      <button onClick={ onClick } className={ styles.triggerBtn }>
        <Icon type="left" />
      </button>
    </div>
  );
};

export default TriggerBtn;
