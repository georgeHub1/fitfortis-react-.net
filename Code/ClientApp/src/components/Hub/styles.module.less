@import "../../styles/_variables.less";

.menuContainer {
  margin: 34px 0;
  min-height: calc(100% - 136px);
  & .logo {
    color: @textPrimary;
    font-weight: 800;
    margin-top: 0;
    font-size: 24px;
    letter-spacing: 0px;
    font-variant-caps: initial;
    overflow: hidden;
    min-width: 27px;
    margin: 20px;
    max-height: 27px;
    transition: width 10s ease-out;
    @media (max-width: 500px) { 
      padding-left: 2.5px;
    }
  }
}

.menuContainer
.menu {
  width: auto;
  height: 100%;
  display: flex;
  padding-top: 5px;
  align-items: center;
  flex-direction: column;
  border-right: none!important;
  background: transparent!important;
  li {
      @media (max-width: 500px) { 
        height: 46px !important;
        padding: 0 5px !important;
      }
     a { 
      span {
        text-overflow: ellipsis;
        overflow: hidden;
      }      
       @media (max-width: 500px) { 
          height: 46px;
        }
        i {
          @media (max-width: 500px) { 
            font-size: 24px !important;
          }
        }
     }
  }
}

.contentContainer {
  margin: 0;
  padding: 24px 45px;
  min-height: auto!important;
  @media (max-width: 767px) { 
    padding: 0 !important;
    background: transparent;
  }
}
@media only screen and (max-device-width: 480px) {
  /* styles for mobile browsers smaller than 480px; (iPhone) */
  .contentContainer {

    padding: 24px 20px;
 
  }
}
.menuContainer .linkClass {
  height: 66px;
  font-weight: 600;
  color: @textPrimary;
  display: flex;
  align-items: center;
}
.MainresponsiveFooter {
  .activeLink {
    position: relative;
    &:before {
      content: "";
      position: absolute;
      left: -12.5px;
      right: 0;
      top: 0;
      bottom: 0;
      border-top: 5px solid;
      width: 50px;
      margin: 0 auto;
    }
    i {
      opacity: 1 !important;
    }
  }
}

.menuContainer .activeLink,
.menuContainer .linkClass:hover {
  background: #335889;
  border-radius: 12px;
  color: #fff;
  height: 100%;
}
.linkClass[disabled]{
  opacity: 0.5;
}

.sideBar {
  display: flex;
}

.menuContainer .navItem {
  height: 66px!important;
  font-weight: 900;
  padding: 0!important;
  margin: 2px 0 0 !important;

  &:active {
    background: none
  }
}

.triggerContainer {
  top: 35px;
  z-index: 9;
  right: -15px;
  position: absolute;
  transition: transform 0.3s ease-out;
}

.triggerBtn {
  width: 30px;
  height: 30px;
  border: none;
  outline: none;
  cursor: pointer;
  border-radius: 50%;
  background: #fff;
  color: @textPrimary;
  box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.15);
}

.triggerBtn:hover {
  color: #fff;
  background: #335889;
}

.rotateContainer {
  transform: rotate(180deg);
}

.avatar {
  width: 34px;
  height: 34px;
  border-radius: 50%;
  margin: 16px 21px !important;
  min-width: 34px!important;
  border: 2px solid #335889;
  @media (max-width: 767px) {
    width: 30px;
    height: 30px;
    margin: 16px 0 !important;
    min-width: 30px!important;
    border-color: #fff;
  }
}
.sidebar_class{ 
  padding: 0 20px;
  background: linear-gradient(to right, #f4f6f7, #f3f5f8) !important;
  @media (max-width: 767px) {
    display: none;
  }
  @media (max-width: 500px) {
    padding: 0px; 
  }
}

.layout_main{
  height: 100%;
}

.backButton{
  min-width: 30px !important;
  width: 30px !important;
  height: 30px !important;
  i {
    color: #0e3459;
  }
}

.width{
  width:30px;
}
.mobileViewHeaderTitle{
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  width: 100%;
  margin: 0 auto 0 15px !important;
  text-align: center;
}
.termsofuselink {
  text-align: center;
  a {
    color: rgba(0, 0, 0, 0.45);
    font-size: 12px;
    cursor: pointer;
    padding: 16px 21px;
    display: inline-block;
    word-break: break-word;
    &:hover {
      color: #507196;
    }
  }
  .toslink {
    color: rgba(0, 0, 0, 0.45);
    font-size: 12px;
    cursor: pointer;
    display: inline-block;
    word-break: break-word;
    padding: 16px 5px;
    &:hover {
      color: #507196;
    }
  }
}
.open.termsofuselink {
  text-align: center;
  a {
    padding: 16px 5px;
  }
}