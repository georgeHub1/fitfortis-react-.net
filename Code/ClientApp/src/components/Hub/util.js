export const getActivePathName = path => {
  if (path.includes('/encyclopedia/')) {
    path = '/encyclopedia';
  }
  if (path.includes('/newsfeed/')) {
    path = '/';
  }

  switch (path) {
    case '/devices': {
      return 'Hub.devices';
    }
    case '/documents': {
      return 'Hub.documents';
    }
    case '/alerts': {
      return 'Hub.alerts';
    }
    case '/metrics': {
      return 'Hub.metrics';
    }
    case '/encyclopedia': {
      return 'Hub.encyclopedia';
    }
    case path.includes('/encyclopedia'): {
      return 'Hub.encyclopedia';
    }
    case '/symptomChecker': {
      return 'Hub.symptomChecker';
    }
    case '/adminTools': {
      return 'Hub.adminTools';
    }
    case '/profile': {
      return 'Hub.profile';
    }
    case '/': {
      return 'Hub.home';
    }
    case '/about': {
      return 'Hub.about';
    }
    case '/about/privacyAndLegal': {
      return 'Hub.privacyAndLegal';
    }
    default: {
      return 'Hub.home';
    }
  }
};
