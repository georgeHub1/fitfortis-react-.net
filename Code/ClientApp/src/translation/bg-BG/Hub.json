{
  "header": "fit fortis",
  "headerMinimized": "ff",

  "adminTools": "За администратори",
  "metrics": "Графики",
  "devices": "Устройства",
  "documents": "Документи",
  "encyclopedia": "Енциклопедия",
  "home": "Начална страница",
  "newsAlerts": "Новини & съобщения",
  "alerts": "Cъобщения",
  "profile": "Профил",
  "symptomChecker": "Диагноза по симптоми",
  "signIn":"Вход",
  "termsofuse": "Условия за ползване",
  "about": "Относно Fitfortis",
  "privacyAndLegal": "Относно Fitfortis"
}
