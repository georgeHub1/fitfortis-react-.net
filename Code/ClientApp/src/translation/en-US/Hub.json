{
  "header": "fit fortis",
  "headerMinimized": "ff",

  "adminTools": "Admin Tools",
  "metrics": "Metrics",
  "devices": "Devices",
  "documents": "Documents",
  "encyclopedia": "Encyclopedia",
  "home": "Home",
  "newsAlerts": "News & Alerts",
  "alerts": "Alerts",
  "profile": "Profile",
  "symptomChecker": "Symptom Checker",
  "signIn":"Sign in",
  "termsofuse": "Terms of use",
  "about": "About Fitfortis",
  "privacyAndLegal": "About Fitfortis"
}
