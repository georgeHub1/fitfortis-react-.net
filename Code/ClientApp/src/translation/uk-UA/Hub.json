{
  "header": "fit fortis",
  "headerMinimized": "ff",

  "adminTools": "Налаштування",
  "metrics": "Метрики",
  "devices": "Пристрої",
  "documents": "Документи",
  "encyclopedia": "Енциклопедія",
  "home": "Головна",
  "newsAlerts": "Новини і сповіщення",
  "alerts": "Cповіщення",
  "profile": "Профіль",
  "symptomChecker": "Аналізатор симптомів",
  "signIn": "Увійти",
  "termsofuse": "Умови користування",
  "about": "Про Fitfortis",
  "privacyAndLegal": "Про Fitfortis"
}
