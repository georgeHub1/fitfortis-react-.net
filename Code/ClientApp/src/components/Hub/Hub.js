import React, { Component } from 'react';
import { NavLink, withRouter, Link } from 'react-router-dom';
import { Layout, Menu, Icon } from 'antd';
import { FormattedMessage } from 'react-intl';
import { connect } from 'react-redux';
import TriggerBtn from './TriggerBtn.js';
import AvatarFemale from '../../img/avatar_Female.svg';
import AvatarMale from '../../img/avatar_Male.svg';
import { ReactComponent as Home } from '../../img/home.svg';
import { ReactComponent as News } from '../../img/news.svg';
import { ReactComponent as Devices } from '../../img/devices.svg';
import { ReactComponent as Metrics } from '../../img/metrics.svg';
import { ReactComponent as Documents } from '../../img/documents.svg';
import { ReactComponent as Encyclopedia } from '../../img/encyclopedia.svg';
import { ReactComponent as SymptomChecker } from '../../img/symptomChecker.svg';
import { ReactComponent as AdminTools } from '../../img/adminTools.svg';
import styles from './styles.module.less';
import { getActivePathName } from './util';

const { Sider, Content, Header, Footer } = Layout;


class Hub extends Component {
  displayName = Hub.name
  constructor (props) {
    super(props);
    this.state = {
      collapsed: false,
      open: false,
      isMobileView: (window.innerWidth <= 767)
    };
  }

  componentWillMount () {
    window.addEventListener('resize', this.resizeEvent);
  }

  componentWillUnmount () {
    window.removeEventListener('resize', this.resizeEvent);
  }

  resizeEvent = e => {
    if (window.innerWidth > 767) {
      this.setState({
        ...this.state,
        isMobileView: false
      });
    } else {
      this.setState({
        ...this.state,
        isMobileView: true
      });
    }
  }

  onCollapse = () => {
    this.setState({ collapsed: !this.state.collapsed });
    this.setState({ open: !this.state.open });
  }

  getActiveClass = () => {
    if (window.location.pathname.includes('newsfeed') || window.location.pathname === '/') {
      return styles.activeLink;
    }
    return '';
  }

  isDisable = () => {
    const { auth } = this.props;

    return !auth.loggenIn;
  }

  isBackButtonRequired = () => {
    const { location } = this.props.history;

    if (location.pathname.includes('/encyclopedia/')) {
      return true;
    }
    if (location.pathname.includes('/newsfeed/')) {
      return true;
    }
    if (location.pathname.includes('/encyclopediaMedicine/')) {
      return true;
    }
    return false;
  }

  goBack = () => {
    const { history } = this.props;

    history.goBack();
  }

  render () {
    const { collapsed, isMobileView, open } = this.state;
    const { auth, account, genderIdentity, avatar, item, userRole } = this.props;
    const defaultAvatar = genderIdentity === 'Female' ? AvatarFemale : AvatarMale;

    return (
      <Layout className={styles.layout_main}>
        {
          !isMobileView
           ? <Sider
          collapsible
          width={ 280 }
          breakpoint='lg'
          trigger={ null }
          collapsedWidth={ 115 }
          className={styles.sidebar_class}
          onCollapse={this.onCollapse}
          collapsed={collapsed} >
          <div className={ styles.menuContainer } style={{ width: '100%' }} >
            <TriggerBtn collapsed={ collapsed } onClick={ this.onCollapse } />
            <div className={ `logo ${styles.logo}` }>
              { collapsed
                ? <FormattedMessage id='Hub.headerMinimized' />
                : <FormattedMessage id='Hub.header' />
              }
            </div>
            <Menu
              selectable={ false }
              className={ styles.menu }
              mode="inline" >
              <Menu.Item className={ styles.navItem } style={{ width: '100%' }}>
                <NavLink
                  activeClassName={ styles.activeLink }
                  className={`${styles.linkClass} ${this.isDisable() ? 'withoutLogin' : ''}`} to="/profile">
                  {
                    auth.loggenIn
                    ? <React.Fragment>

                      <img className={ `${ styles.avatar } anticon` } src={avatar ? avatar :  defaultAvatar } alt='avatar' />
                      <span>{account.firstName}</span>
                    </React.Fragment>
                    : <React.Fragment>
                      <Icon style={{ fontSize: '34px', height: '34px', margin: '16px 21px' }} type="login" />
                      <FormattedMessage id='Hub.signIn' tagName='span'/>
                    </React.Fragment>
                  }
                </NavLink>
              </Menu.Item>
              <Menu.Item className={ styles.navItem } style={{ width: '100%' }}>
                <NavLink
                  exact
                  className={`${styles.linkClass} ${this.getActiveClass()}`}
                  to="/"
                  >
                  <Icon style={{ fontSize: '34px', height: '34px', margin: '16px 21px' }} component={ Home } />
                  <FormattedMessage id='Hub.home' tagName='span'/>
                </NavLink>
              </Menu.Item>
              <Menu.Item className={ styles.navItem } style={{ width: '100%' }}>
                <NavLink
                  activeClassName={ styles.activeLink }
                  className={`${styles.linkClass} ${this.isDisable() ? 'withoutLogin' : ''}`}
                  to="/metrics">
                  <Icon style={{ fontSize: '34px', height: '34px', margin: '16px 21px' }} component={ Metrics } />
                  <FormattedMessage id='Hub.metrics' tagName='span'/>
                </NavLink>
              </Menu.Item>
              <Menu.Item className={ styles.navItem } style={{ width: '100%' }}>
                <NavLink
                  activeClassName={ styles.activeLink }
                  className={`${styles.linkClass} ${this.isDisable() ? 'withoutLogin' : ''}`}
                  to="/alerts">
                  <Icon style={{ fontSize: '34px', height: '34px', margin: '16px 21px' }} component={ News }  />
                  <FormattedMessage id='Hub.alerts' tagName='span'/>
                </NavLink>
              </Menu.Item>
              {
                <Menu.Item className={ styles.navItem } style={{ width: '100%' }}>
                  <NavLink
                    activeClassName={ styles.activeLink }
                    className={`${styles.linkClass} ${this.isDisable() ? 'withoutLogin' : ''}`}
                    to="/documents">
                    <Icon style={{ fontSize: '34px', height: '34px', margin: '16px 21px' }} component={ Documents }  />
                    <FormattedMessage id='Hub.documents' tagName='span'/>
                  </NavLink>
                </Menu.Item>
              }
              {
                userRole.toLowerCase() !== 'user'
                && <Menu.Item className={ styles.navItem } style={{ width: '100%' }}>
                  <NavLink
                    disabled={this.isDisable()}
                    activeClassName={ styles.activeLink }
                    className={ styles.linkClass } to="/devices">
                    <Icon style={{ fontSize: '34px', height: '34px', margin: '16px 21px' }} component={ Devices }  />
                    <FormattedMessage id='Hub.devices' tagName='span'/>
                  </NavLink>
                </Menu.Item>
              }
              <Menu.Item className={ styles.navItem } style={{ width: '100%' }}>
                <NavLink
                  activeClassName={ styles.activeLink }
                  className={window.location.pathname.includes('encyclopediaMedicine') ? `${styles.linkClass} ${styles.activeLink}` : styles.linkClass }
                  to="/encyclopedia">
                  <Icon style={{ fontSize: '34px', height: '34px', margin: '16px 21px' }} component={ Encyclopedia }  />
                  <FormattedMessage id='Hub.encyclopedia' tagName='span'/>
                </NavLink>
              </Menu.Item>
              {
                userRole.toLowerCase() !== 'user'
                && <Menu.Item className={ styles.navItem } style={{ width: '100%' }}>
                  <NavLink
                    disabled={this.isDisable()}
                    activeClassName={ styles.activeLink }
                    className={ styles.linkClass } to="/symptomChecker">
                    <Icon style={{ fontSize: '34px', height: '34px', margin: '16px 21px' }} component={ SymptomChecker }  />
                    <FormattedMessage id='Hub.symptomChecker' tagName='span'/>
                  </NavLink>
                </Menu.Item>
              }
              {
                userRole.toLowerCase() !== 'user'
                ? <Menu.Item className={ styles.navItem } style={{ width: '100%' }}>
                  <NavLink
                    disabled={this.isDisable()}
                    activeClassName={ styles.activeLink }
                    className={ styles.linkClass } to="/adminTools">
                    <Icon style={{ fontSize: '34px', height: '34px', margin: '16px 21px' }} component={ AdminTools } />
                    <FormattedMessage id='Hub.adminTools' tagName='span'/>
                  </NavLink>
                </Menu.Item>
                : ''
              }
            </Menu>
          </div>
          <div className={styles.termsofuselink + (open ? ' open' : '')}>
            <Link className={styles.toslink} to='/about'>
              <FormattedMessage id='Hub.termsofuse' tagName='span'/>
            </Link>
          </div>
        </Sider>
        : <HeaderComponent
          location={this.props.location}
          account={account}
          defaultAvatar={defaultAvatar}
          avatar={avatar}
          isBackButton={this.isBackButtonRequired()}
          goBack={this.goBack}
          item={item}
          isDisable={this.isDisable()}
        />
      }
        <Layout
          id="main_container"
          breakpoint='md'
          className={`main_layout_main_inner ${styles.layout_main_inner}`}
        >
          <Content className={ `${styles.contentContainer} ${styles.responsivecontentContainer}` }>
            {this.props.children}
          </Content>
        </Layout>
        {
          !isMobileView
          ? ''
          : <FooterComponent getActiveClass={this.getActiveClass} userRole={userRole} isDisable={this.isDisable()}/>
        }
      </Layout>
    );
  }
}

const HeaderTitle = ({ item }) => {
  return (item ? <h1 className={styles.mobileViewHeaderTitle}>{item.title}</h1> : '');
};


const HeaderComponent = ({ item, location, account, defaultAvatar, avatar, isBackButton, goBack, isDisable }) => {
  return (
    <Header className="responsiveHeader" theme={'light'} >
      <div className="innerResponsiveHeader">
        <React.Fragment>
          <NavLink
            aria-label="profile tab"
            activeClassName={ styles.activeLink }
            className={`${styles.linkClass}`}
            to="/profile">
             <img className={ `${ styles.avatar } anticon` } src={avatar ? avatar :  defaultAvatar } alt='avatar' />
           </NavLink>
        </React.Fragment>
        {
          isBackButton
          ? <HeaderTitle item={item} />
          : <FormattedMessage id={getActivePathName(location.pathname)} tagName='h1'/>
         }
        <React.Fragment>
          <NavLink
            aria-label="alerts tab"
            activeClassName={ styles.activeLink }
            className={`${styles.linkClass}`}
            to="/alerts">
            <Icon style={{ fontSize: '30px', height: '30px' }} component={ News }  />
          </NavLink>
        </React.Fragment>
      </div>
    </Header>
  );
};

const FooterComponent = ({ getActiveClass, userRole, isDisable }) => {
  return (
     <Footer className= {`responsiveFooter ${styles.MainresponsiveFooter}`}>
     <div className="innerResponsiveFooter">
      <NavLink
      aria-label="hmoe tab"
       exact
       activeClassName={ styles.activeLink }
       className={`${styles.linkClass} ${getActiveClass()}` }
       to="/">
        <Icon component={ Home } />
      </NavLink>
       <NavLink
       aria-label="metrics tab"
        activeClassName={ styles.activeLink }
        className={`${styles.linkClass}`}
        to="/metrics">
        <Icon component={ Metrics } />
      </NavLink>
      {
        <NavLink
        aria-label="documents tab"
          activeClassName={ styles.activeLink }
          className={`${styles.linkClass}`}
          to="/documents">
          <Icon component={ Documents }  />
        </NavLink>
      }
      {
        userRole.toLowerCase() !== 'user'
        && <NavLink
          aria-label="devices tab"
          activeClassName={ styles.activeLink }
          className={ styles.linkClass } to="/devices">
          <Icon component={ Devices }  />
        </NavLink>
      }
      <NavLink
        aria-label="encyclopedia tab"
        activeClassName={ styles.activeLink }
        className={window.location.pathname.includes('encyclopediaMedicine') ? `${styles.linkClass} ${styles.activeLink}` : styles.linkClass }
        to="/encyclopedia">
        <Icon component={ Encyclopedia }  />
      </NavLink>
      {
        userRole.toLowerCase() !== 'user'
        && <NavLink
          aria-label="symptomChecker tab"
          activeClassName={ styles.activeLink }
          className={ styles.linkClass } to="/symptomChecker">
          <Icon component={ SymptomChecker }  />
        </NavLink>
      }
      </div>
     </Footer>
  );
};

const mapStateToProps = ({ auth, item, profile: { account, userRole, basicInformation: { genderIdentity }, avatar } }) => ({ auth, account, genderIdentity, avatar, item, userRole });

export default connect(
  mapStateToProps
)(withRouter(Hub));
