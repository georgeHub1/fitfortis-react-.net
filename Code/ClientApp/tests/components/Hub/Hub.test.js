import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Hub from '../../../src/components/Hub/Hub.js';
import createAppStore from '../../../src/redux/';

const Container = Hub;

beforeEach(() => {
  const consoleErrorMock = jest.fn();

  console.error = consoleErrorMock;
});
describe('Hub', () => {
  it('renders correctly', () => {
    const wrapper = renderer
      .create(
        <MemoryRouter>
          <Provider store={createAppStore}>
            <Container />
          </Provider>
        </MemoryRouter>
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
