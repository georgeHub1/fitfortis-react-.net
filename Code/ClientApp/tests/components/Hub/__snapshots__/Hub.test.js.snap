// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hub renders correctly 1`] = `
<section
  className="ant-layout ant-layout-has-sider"
>
  <aside
    className="ant-layout-sider ant-layout-sider-dark"
    style={
      Object {
        "flex": "0 0 280px",
        "maxWidth": "280px",
        "minWidth": "280px",
        "width": "280px",
      }
    }
  >
    <div
      className="ant-layout-sider-children"
    >
      <div
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <div
          className=" undefined"
        >
          <button
            onClick={[Function]}
          >
            <i
              aria-label="icon: left"
              className="anticon anticon-left"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="left"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                />
              </svg>
            </i>
          </button>
        </div>
        <div
          className="logo undefined"
        />
        <ul
          className="ant-menu ant-menu-light ant-menu-root ant-menu-inline"
          onMouseEnter={[Function]}
          onTransitionEnd={[Function]}
          role="menu"
          style={Object {}}
        >
          <li
            className="ant-menu-item"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={
              Object {
                "paddingLeft": 24,
                "width": "100%",
              }
            }
          >
            <a
              aria-current={null}
              className="undefined withoutLogin"
              href="/profile"
              onClick={[Function]}
            >
              <i
                aria-label="icon: login"
                className="anticon anticon-login"
                style={
                  Object {
                    "fontSize": "34px",
                    "height": "34px",
                    "margin": "16px 21px",
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="login"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M521.7 82c-152.5-.4-286.7 78.5-363.4 197.7-3.4 5.3.4 12.3 6.7 12.3h70.3c4.8 0 9.3-2.1 12.3-5.8 7-8.5 14.5-16.7 22.4-24.5 32.6-32.5 70.5-58.1 112.7-75.9 43.6-18.4 90-27.8 137.9-27.8 47.9 0 94.3 9.3 137.9 27.8 42.2 17.8 80.1 43.4 112.7 75.9 32.6 32.5 58.1 70.4 76 112.5C865.7 417.8 875 464.1 875 512c0 47.9-9.4 94.2-27.8 137.8-17.8 42.1-43.4 80-76 112.5s-70.5 58.1-112.7 75.9A352.8 352.8 0 0 1 520.6 866c-47.9 0-94.3-9.4-137.9-27.8A353.84 353.84 0 0 1 270 762.3c-7.9-7.9-15.3-16.1-22.4-24.5-3-3.7-7.6-5.8-12.3-5.8H165c-6.3 0-10.2 7-6.7 12.3C234.9 863.2 368.5 942 520.6 942c236.2 0 428-190.1 430.4-425.6C953.4 277.1 761.3 82.6 521.7 82zM395.02 624v-76h-314c-4.4 0-8-3.6-8-8v-56c0-4.4 3.6-8 8-8h314v-76c0-6.7 7.8-10.5 13-6.3l141.9 112a8 8 0 0 1 0 12.6l-141.9 112c-5.2 4.1-13 .4-13-6.3z"
                  />
                </svg>
              </i>
            </a>
          </li>
          <li
            className="ant-menu-item"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={
              Object {
                "paddingLeft": 24,
                "width": "100%",
              }
            }
          >
            <a
              aria-current="page"
              className="undefined undefined active"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              <i
                className="anticon"
                style={
                  Object {
                    "fontSize": "34px",
                    "height": "34px",
                    "margin": "16px 21px",
                  }
                }
              >
                <svg
                  aria-hidden={true}
                  className=""
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  width="1em"
                >
                  home.svg
                </svg>
              </i>
            </a>
          </li>
          <li
            className="ant-menu-item"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={
              Object {
                "paddingLeft": 24,
                "width": "100%",
              }
            }
          >
            <a
              aria-current={null}
              className="undefined withoutLogin"
              href="/metrics"
              onClick={[Function]}
            >
              <i
                className="anticon"
                style={
                  Object {
                    "fontSize": "34px",
                    "height": "34px",
                    "margin": "16px 21px",
                  }
                }
              >
                <svg
                  aria-hidden={true}
                  className=""
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  width="1em"
                >
                  metrics.svg
                </svg>
              </i>
            </a>
          </li>
          <li
            className="ant-menu-item"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={
              Object {
                "paddingLeft": 24,
                "width": "100%",
              }
            }
          >
            <a
              aria-current={null}
              className="undefined withoutLogin"
              href="/alerts"
              onClick={[Function]}
            >
              <i
                className="anticon"
                style={
                  Object {
                    "fontSize": "34px",
                    "height": "34px",
                    "margin": "16px 21px",
                  }
                }
              >
                <svg
                  aria-hidden={true}
                  className=""
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  width="1em"
                >
                  news.svg
                </svg>
              </i>
            </a>
          </li>
          <li
            className="ant-menu-item"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={
              Object {
                "paddingLeft": 24,
                "width": "100%",
              }
            }
          >
            <a
              aria-current={null}
              className="undefined withoutLogin"
              href="/documents"
              onClick={[Function]}
            >
              <i
                className="anticon"
                style={
                  Object {
                    "fontSize": "34px",
                    "height": "34px",
                    "margin": "16px 21px",
                  }
                }
              >
                <svg
                  aria-hidden={true}
                  className=""
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  width="1em"
                >
                  documents.svg
                </svg>
              </i>
            </a>
          </li>
          <li
            className="ant-menu-item"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={
              Object {
                "paddingLeft": 24,
                "width": "100%",
              }
            }
          >
            <a
              aria-current={null}
              href="/encyclopedia"
              onClick={[Function]}
            >
              <i
                className="anticon"
                style={
                  Object {
                    "fontSize": "34px",
                    "height": "34px",
                    "margin": "16px 21px",
                  }
                }
              >
                <svg
                  aria-hidden={true}
                  className=""
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  width="1em"
                >
                  encyclopedia.svg
                </svg>
              </i>
            </a>
          </li>
        </ul>
      </div>
      <div
        className="undefined"
      >
        <a
          href="/about"
          onClick={[Function]}
        />
      </div>
    </div>
  </aside>
  <section
    breakpoint="md"
    className="main_layout_main_inner undefined ant-layout"
    id="main_container"
  >
    <main
      className="undefined undefined ant-layout-content"
    />
  </section>
  
</section>
`;
